relay-filter/src/config.rs

```diff
@@ -416,34 +409,60 @@ pub struct GenericFiltersConfig {
     ///
     /// The map contains unique filters, meaning there are no two filters with
     /// the same id. See struct docs for more details.
-    #[serde(
-        default,
-        skip_serializing_if = "IndexMap::is_empty",
-        with = "generic_filters_custom_serialization"
-    )]
-    pub filters: IndexMap<String, GenericFilterConfig>,
+    #[serde(default, skip_serializing_if = "GenericFiltersMap::is_empty")]
+    pub filters: GenericFiltersMap,
 }
 
 impl GenericFiltersConfig {
     /// Returns true if the filters are not declared.
     pub fn is_empty(&self) -> bool {
-        self.filters.is_empty()
+        self.filters.0.is_empty()
     }
 }
 
-mod generic_filters_custom_serialization {
-    use super::*;
+/// Map of generic filters, mapping from the id to the filter itself.
+#[derive(Clone, Debug, Default)]
+pub struct GenericFiltersMap(IndexMap<String, GenericFilterConfig>);
+
+impl GenericFiltersMap {
+    /// Returns an empty map.
+    pub fn new() -> Self {
+        GenericFiltersMap(IndexMap::new())
+    }
+
+    /// Returns whether the map is empty.
+    pub fn is_empty(&self) -> bool {
```

Comment:
With the deref this shouldnt be necessary

Reply:
Needed for serde annotation

Reply:
It's used for skip serialization in serde.